core/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -102,6 +104,7 @@ protected boolean removeEldestEntry(Map.Entry eldest) {
 
     public TransportService(Transport transport, ThreadPool threadPool) {
         this(EMPTY_SETTINGS, transport, threadPool);
+        setTaskManager(new TaskManager(EMPTY_SETTINGS));
```

Comment:
Is this constructor just for testing or is it used somewhere else? Maybe a comment about why this is ok here.....